dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/LayoutAttributeTests.java

```diff
@@ -0,0 +1,382 @@
+            assertTrue("table element should have been present", page.asXml().contains("<table"));
+            assertFalse("ul element should NOT have been present", page.asXml().contains("<ul"));
+
+            //Specific assertions input
+            assertTrue("Element with ID form:input should have been a table", page.getElementById("form:input") instanceof HtmlTable);
+        } else {
+            //General assertions
+            assertFalse("table element should NOT have been present", page.asXml().contains("<table"));
+            assertTrue("ul element should have been present", page.asXml().contains("<ul"));
+
+            //Specific assertions input
+            assertTrue("Element with ID form:input should have been an unordered list", page.getElementById("form:input") instanceof HtmlUnorderedList);
+        }
+
+        if (assertBorder) {
+            assertFalse("Element with ID form:input should have had a border attribute", page.getElementById("form:input").getAttribute("border").isBlank());
+            assertEquals("Element with ID form:input should have had a border with value: " + borderPixels.intValue() + " instead of "
+                         + page.getElementById("form:input").getAttribute("border"), borderPixels.intValue(),
+                         Integer.parseInt(page.getElementById("form:input").getAttribute("border")));
+        } else {
+            assertTrue("Element with ID form:input should NOT have had a border attribute", page.getElementById("form:input").getAttribute("border").isEmpty());
+        }
+
+        //Specific assertions for input type
+        String formId = "form:input:" + items.indexOf(testItem);
+        assertNotNull("Element with ID form:input:" + formId + " should exist", page.getElementById(formId));
+        assertTrue("Element with ID form:input:" + formId + " should have type " + InputType.getSimpleName(), InputType.isInstance(page.getElementById(formId)));
+        assertNotNull("Element with ID form:input:" + formId + " should have a value attribute", page.getElementById(formId).getAttribute("value"));
+        assertEquals("Element with ID form:input:" + formId + " should have a value of " + testItem.getId() + " but was " + page.getElementById(formId).getAttribute("value"),
+                     testItem.getId(),
+                     Long.parseLong(page.getElementById(formId).getAttribute("value")));
+    }
+
+    private boolean writeOutputToFile(String xmlContent, String fileName) {
+        //Re-enable for easier debugging
+//        File outputDir = new File(server.getLogsRoot(), "output");
+//        File outputFile = new File(outputDir, fileName);
+//        outputDir.mkdirs();
+//
+//        Log.info(c, "writeOutputToFile", outputFile.getAbsolutePath());
+//        try (FileOutputStream fos = new FileOutputStream(outputFile, true)) {
+//            fos.write(xmlContent.getBytes());
+//        } catch (Exception e) {
+//            return false;
+//        }
```

Comment:
I was this locally and this is pretty convenient!  One thing -- could you use the full path reference for File and FileOutputStream (imports aren't here) so it can just be un-commented and ready to us?

Reply:
Decided to just move this to FATSuite and only run it if we are running locally.